lib/mailjet/mailer.rb

```diff
@@ -27,25 +27,181 @@ def initialize(options = {})
 # The deliver methods maps the Mail::Message attributes to the MailjetSend API JSON expected structure
 class Mailjet::APIMailer
   def initialize(options = {})
-    @delivery_method_options = options.slice(
+#    send = Mailjet.Send.new
+#    if send.version.exists
+#      version = send.version
+#    else
+      @version = Mailjet.config.api_version
+#    end
+    @delivery_method_options_v3_0 = options.slice(
       :recipients, :'mj-prio', :'mj-campaign', :'mj-deduplicatecampaign',
       :'mj-templatelanguage', :'mj-templateerrorreporting', :'mj-templateerrordeliver', :'mj-templateid',
       :'mj-trackopen', :'mj-trackclick',
-      :'mj-customid', :'mj-eventpayload', :vars, :headers
+      :'mj-customid', :'mj-eventpayload', :vars, :headers,
+    )
+    @delivery_method_options_v3_1 = options.slice(
+      :recipients, :'prio', :'campaign', :'deduplicatecampaign',
+      :'templatelanguage', :'templateerrorreporting', :'templateerrordeliver', :'templateid',
+      :'trackopen', :'trackclick',
+      :'customid', :'eventpayload', :vars, :headers,
     )
   end
 
-  def deliver!(mail)
+  def deliver!(mail, options = nil)
+#    p mail.header.fields
+    
+    if (options && options.kind_of?(Object) && options['version'].present?)
+      @version = options['version']
+    end
+    
+    if (!options.kind_of?(Object))
```

Comment:
Move that test up. You're already doing it at line https://github.com/mailjet/mailjet-gem/pull/87/files#diff-1713bdd78158071c925dcd84baed4df2R53